lib/migration.pm

```diff
@@ -131,10 +131,12 @@ sub check_rollback_system {
     return unless is_sle;
     # Check SUSEConnect status for SLE
     # check rollback-helper service is enabled and worked properly
-    # Wait some time if rollback is in activating state
-    # Consider a bug if rollback service isn't done after that time
-    script_run('for x in `seq 60`; do (systemctl --no-pager status rollback | grep -q activating) && sleep 5; done');
```

Comment:
what was wrong with the bash approach? 60*5 should be same as 5*60 so why change this?

Reply:
@okurz It's difference , sleep in the bash, it only let the guest VM wait, but can not stop test progress, so that it still cause case failed, please see following link in OSD, https://openqa.suse.de/tests/1689418#step/snapper_rollback/39. 

The new method will not stop test, add wait in a loop, like you said, it will exit after 5 tries even if the exit code is not 0, but it should be a bug which not start rollback service in 5 minutes.

Reply:
Ok, in https://openqa.suse.de/tests/1689418#step/snapper_rollback/39 one can not easily see how the for-loop exits, if it was actually waiting or not. This is also why I would recommend to not use script_run and ignore the exit code. However, in https://openqa.suse.de/tests/1689418/file/serial0.txt you can see "jMe0F-141-" on the serial line so the return code of the for-loop execution is "141", not "0" and not "1", what you would expect from grep. I guess what happens here is that due to the option "pipefail" in our bash sessions the code does not work as expected. The "141" points to a SIGPIPE which can happen because the grep returns successfully but then terminating the systemctl status call with SIGPIPE causing the "&& sleep 5" to be never executed and therefore the loop exits way faster without the waiting time you expect.

Check for yourself and compare

```bash
bash -exo pipefail -c "for i in {1..100} ; do (sudo systemctl status dbus | grep -q active) && sleep 5; done"
```

versus

```bash
bash -ex -c "for i in {1..100} ; do (sudo systemctl status dbus | grep -q active) && sleep 5; done"
```

The former returns 141 without sleeping (maybe sleeping only once), the latter sleeps 5 seconds from the beginning.

So to repeat: Please use bash, please don't ignore the return code of script_run, better use assert_script_run

Reply:
I have tried the two bash on my local. The first bash does not work even if the service active, like you said , return 141, and it keep running until for-loop ended. The second bash can exit with code 0 when the service active and can run `sleep 5` in the guest, but it also like the first one, keep running until for-loop ended,So that, it can not exit immediately when service active. So, I think they are both not accept. 
For my new method, to use `script_run` to check the rollback service whether active or not, there has two reasons. First  is we need the output of rollback service status based on https://bugzilla.suse.com/show_bug.cgi?id=1089478#c5. Second is we need more time to wait service active in some cases with more modules, so that, we can not use `assert_script_run`, we only try to wait, we also check the exit code to exit for-loop, and add `systemctl('is-active rollback')` to check  service status to decide this case fail or pass.

Reply:
> So, I think they are both not accept.

of course. You can put in a "break" to stop whenever the grep did not find "activating".

Reply:
yes, it can break when did not found "activating", it is the same result as my new method with checking exit code. and I think the bash way is too overage, because it also need to check the output of rollback service with another code - `systemctl('status rollback')`. but my for-loop does not need again. So, I think mine is better..